reference/matrix/dense_kernels.cpp

```diff
@@ -374,11 +374,15 @@ void compute_mean(std::shared_ptr<const ReferenceExecutor> exec,
                   const matrix::Dense<ValueType>* x,
                   matrix::Dense<ValueType>* result, array<char>&)
 {
+    GKO_ASSERT_EQ(result->get_size()[0], 1);
```

Comment:
I think we (mostly) don't throw in our backend libraries. You could just add this check to the core file.